### PR TITLE
ui: Create auth-method show page with General Info Tab

### PR DESCRIPTION
**Pull request by kaxcode**
✨  Creates the show page for auth-methods of type Kubernetes, OIDC, and JWT.

[Visit demo
](https://consul-ui-staging-kslwven8x-hashicorp.vercel.app/ui/dc1/services#CONSUL_ACLS_ENABLE=1)
auth-method-show-oidc

### CODE REVIEW

**Comment by kaxcode** [comment]
@johncowen ready for review 😄

File: ui/packages/consul-ui/app/components/certificate/index.hbs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<div class="certificate">
+  <CopyButton @value={{@item}} @name={{@name}} />
+  <button
+    type="button"
+    class={{concat "visibility" (if this.show " hide" " show")}}
+    {{on "click" this.setVisibility}}
```
Comment (johncowen) [inline ui/packages/consul-ui/app/components/certificate/index.hbs:6]: Could we use the `set` helper here?

File: ui/packages/consul-ui/app/components/certificate/index.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class TopologyMetrics extends Component {
```
Comment (johncowen) [inline ui/packages/consul-ui/app/components/certificate/index.js:5]: If you don't want to go with the `set` helper the name of this class should be `Certificate`.

File: ui/packages/consul-ui/app/components/certificate/index.scss [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    @extend %with-visibility-show-icon, %as-pseudo;
+  }
+  div.key {
+    background-color: var(--gray-050);
+    overflow-wrap: anywhere;
+  }
+  hr {
+    border: 3px dashed var(--gray-300);
+    background-color: $white;
+    width: 10vw;
```
Comment (johncowen) [inline ui/packages/consul-ui/app/components/certificate/index.scss:22]: No biggie, wondered if this should be a fixed width so it doesn't change when I resize my window?

File: ui/packages/consul-ui/app/components/consul/auth-method/index.scss [hunk 4]
```diff
@@ -1,9 +1,74 @@
...
+    flex-wrap: wrap;
+    > dt:last-of-type,
+    > dd:last-of-type {
+      border-bottom: 1px solid $gray-300 !important;
+    }
+    dt, dd {
+      padding: 12px 0;
+      margin: 0;
+      border-top: 1px solid $gray-300 !important;
+      color: $black !important;
```
Comment (johncowen) [inline ui/packages/consul-ui/app/components/consul/auth-method/index.scss:45]: I think we decided that we were going to move all these grays to use CSS properties instead of SASS vars, then we don't have a mix of `$gray-***` and `var(--gray-***)`

File: ui/packages/consul-ui/app/components/consul/auth-method/view/index.hbs [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+      <dt>CA Cert</dt>
+      <dd>
+        <Certificate @item={{@item.Config.CACert}} @name={{t 'models.auth-method.Config.CACert'}} />
+      </dd>
+      {{/if}}
+      {{#if @item.Config.ServiceAccountJWT}}
+      <dt>Service account JSON Web Token</dt>
+      <dd>
+        <CopyButton @value={{@item.Config.ServiceAccountJWT}} @name="Service account JSON Web Token" />
+        <span>
```
Comment (johncowen) [inline ui/packages/consul-ui/app/components/consul/auth-method/view/index.hbs:30]: I think there're some `t`s missed here, not sure if you meant to do all these or not? Seems like sometimes `t` is used and sometimes not?

File: ui/packages/consul-ui/app/components/consul/auth-method/view/index.hbs [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+            <td>{{get entry 0}}</td>
+            <td>{{get entry 1}}</td>
+          </tr>
+          {{/each}}
+        </tbody>
+      </table>
+    </section>
+    {{/if}}
+  {{/if}}
+  </div>
```
Comment (johncowen) [inline ui/packages/consul-ui/app/components/consul/auth-method/view/index.hbs:217]: I think these two sections need empty states when they are empty. Would be good for another PR though.

File: ui/packages/consul-ui/mock-api/v1/acl/auth-method/_ [hunk 7]
```diff
@@ -14,17 +27,45 @@ ${
       case 'oidc':
         config = {
           OIDCDiscoveryURL: `https://${fake.internet.ip()}:8443`,
+          OIDCDiscoveryCACert: `-----BEGIN CERTIFICATE-----${fake.internet.password(1357)}-----END CERTIFICATE-----`,
```
Comment (johncowen) [inline ui/packages/consul-ui/mock-api/v1/acl/auth-method/_:30]: Do we know the format of how these come out of consul in the API?

'Generally' they have returns in them (which might mean you don't need to do the hard wrapping in the CSS to show these). If they come out without the returns then don't worry!

File: ui/packages/consul-ui/translations/en-us.yaml [hunk 8]
```diff
@@ -180,3 +180,35 @@ components:
...
+      JWKSURL: JWKS URL
+      JWKSCACert: JWKS CA Cert
+      JWTValidationPubKeys: JWT validation pub keys
+      OIDCDiscoveryURL: Discovery URL
+      JWTSupportedAlgs: JWT supported algorithms
+      BoundAudiences: Bound audiences
+      BoundIssuer: Bound issuer
+      ExpirationLeeway: Expiration leeway
+      NotBeforeLeeway: Not before leeway
+      ClockSkewLeeway: Clocks skew leeway
```
Comment (johncowen) [inline ui/packages/consul-ui/translations/en-us.yaml:203]:
```suggestion
      ClockSkewLeeway: Clock skew leeway
```

**Review by johncowen — APPROVED** [review]
Thanks for the Demo Link, I tried it out and all looks good!

I left a few minor things, if you wanted to look at those before merging, but otherwise all good!

File: ui/packages/consul-ui/app/components/certificate/index.scss (hunk 3 above)
Comment (kaxcode) [reply ui/packages/consul-ui/app/components/certificate/index.scss:22]: 💯

File: ui/packages/consul-ui/mock-api/v1/acl/auth-method/_ (hunk 7 above)
Comment (kaxcode) [reply ui/packages/consul-ui/mock-api/v1/acl/auth-method/_:30]: I assume the same way certs come from other endpoints. This is  why I'm mocking it out the same way.

File: ui/packages/consul-ui/app/components/consul/auth-method/view/index.hbs (hunk 5 above)
Comment (kaxcode) [reply ui/packages/consul-ui/app/components/consul/auth-method/view/index.hbs:30]: Whoops! Looks like I forgot to fix up the 'kubernetes' section.

File: ui/packages/consul-ui/app/components/certificate/index.hbs (hunk 1 above)
Comment (kaxcode) [reply ui/packages/consul-ui/app/components/certificate/index.hbs:6]: I'll keep the action.
